src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/Process_StreamTests.cs

```diff
@@ -116,19 +134,89 @@ public void Process_SyncStreams()
         [Fact]
         public void Process_AsyncHalfCharacterAtATime()
         {
-            var receivedOutput = false;
+            var receivedOutput = new ManualResetEventSlim(false);
             Process p = CreateProcessByteAtATime();
             p.StartInfo.RedirectStandardOutput = true;
             p.StartInfo.StandardOutputEncoding = Encoding.Unicode;
             p.OutputDataReceived += (s, e) =>
             {
                 Assert.Equal(e.Data, "a");
-                receivedOutput = true;
+                receivedOutput.Set();
             };
             p.Start();
             p.BeginOutputReadLine();
             Assert.True(p.WaitForExit(WaitInMS));
-            Assert.True(receivedOutput);
+            Assert.True(receivedOutput.Wait(WaitInMS));
+        }
+
+        [Fact]
+        public void Process_StreamNegativeTests()
+        {
+            {
+                Process p = new Process();
+                Assert.Throws<InvalidOperationException>(() => p.StandardOutput);
+                Assert.Throws<InvalidOperationException>(() => p.StandardError);
+                Assert.Throws<InvalidOperationException>(() => p.BeginOutputReadLine());
+                Assert.Throws<InvalidOperationException>(() => p.BeginErrorReadLine());
+                Assert.Throws<InvalidOperationException>(() => p.CancelOutputRead());
+                Assert.Throws<InvalidOperationException>(() => p.CancelErrorRead());
+            }
+
+            {
+                Process p = CreateProcessStream();
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.RedirectStandardError = true;
+                p.OutputDataReceived += (s, e) => {};
+                p.ErrorDataReceived += (s, e) => { };
+
+                p.Start();
+                p.BeginOutputReadLine();
+                p.BeginErrorReadLine();
+
+                Assert.Throws<InvalidOperationException>(() => p.StandardOutput);
+                Assert.Throws<InvalidOperationException>(() => p.StandardError);
+
+                if (p.WaitForExit(WaitInMS))
+                    p.WaitForExit(); // This ensures async event handlers are finished processing.
+            }
+
+            {
+                Process p = CreateProcessStream();
```

Comment:
I must just be missing something... is this block not identical to the previous one?